controllers/crds/cninode_controller.go

```diff
@@ -276,17 +251,60 @@ func (r *CNINodeReconciler) createCNINodeFromObj(ctx context.Context, newCNINode
 		})
 }
 
-func (r *CNINodeReconciler) GetNodeID(ctx context.Context, nodeName string) (string, error) {
-	node := &v1.Node{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: nodeName}, node)
-	if err != nil {
-		return "", fmt.Errorf("failed to get the node object %s node, err: %s", nodeName, err)
+func (r *CNINodeReconciler) ensureTagsAndLabels(cniNode *v1alpha1.CNINode, node *v1.Node, nodeFound bool) (bool, error) {
```

Comment:
I think the `nodeFound` is unnecessary since also passing node. Not sure why we pass node as pointer, but we can always check if the node is constructed.